TextFormat.md

```diff
@@ -43,43 +38,377 @@ support more human-readable representations, but never at the cost of accurate r
+works in all block-like constructs (`block`, function body, `if`, etc.)
+
+`>s` means *signed* greater-than. explicit unsigned or signed operators will be
+suffixed with 'u' or 's', respectively.
+
+The `$` sigil on user names cleanly ensures that they never collide with wasm
+keywords, present or future.
+
+The last statement has no trailing semicolon; this indicates that its result
+value is being used as the block result value.
+
+
+### Linear memory addresses
+
+Addresses are printed as `[base,+offset,align=val]`. The offset can be omitted
+when it is zero, and the alignment can be omitted for natural alignment.
+
+Addresses are prefixed by an identifier giving the type of a load, or the
+type of the value to store, optionally followed by a colon and a number of
+bits of memory to access (for truncating stores or extending loads).
+For loads, the number of bits is followed by `s` or `u` for signed and
+unsigned.
+
+Examples:
+
+```
+  function $loads_and_stores ($ptr : i32) : (i32) {
+    var $t : i32;
+
+    // Simple cases.
+    $t = i32[$ptr];           // plain i32.load from address $ptr
+    i32[$ptr] = $t;           // plain i32.store of $t to address $ptr
+
+    // Various modifiers.
+    f32[$ptr,+4] = -0x1.8p0;  // f32.store to address $ptr plus 4
+    i64:8s[$ptr,+17,align=1]; // i64.load8_s from address $ptr plus 17 with alignment 1
+  }
+```
+
+### A slightly larger example:
+
+Here's some C code:
+
+```
+  float Q_rsqrt(float number)
+  {
+      long i;
+      float x2, y;
+      const float threehalfs = 1.5F;
+
+      x2 = number * 0.5F;
+      y  = number;
+      i  = *(long *) &y;
+      i  = 0x5f3759df - (i >> 1);
+      y  = *(float *) &i;
+      y  = y * (threehalfs - (x2 * y * y));
+      y  = y * (threehalfs - (x2 * y * y));
+
+      return y;
+  }
+```
+
+Here's the corresponding LLVM wasm backend output + binaryen + slight tweaks:
+
+```
+  (func $Q_rsqrt (param $0 f32) (result f32)
+    (local $1 f32)
+    (set_local $1
+      (f32.reinterpret/i32
+        (i32.sub
+          (i32.const 1597463007)
+          (i32.shr_s
+            (i32.reinterpret/f32
+              (get_local $0))
+            (i32.const 1)))))
+    (set_local $1
+      (f32.mul
+        (get_local $1)
+        (f32.sub
+          (f32.const 0x1.8p+0)
+          (f32.mul
+            (get_local $1)
+            (f32.mul
+              (get_local $1)
+              (set_local $0
+                (f32.mul
+                  (get_local $0)
+                  (f32.const 0x1p-1))))))))
+    (f32.mul
+      (get_local $1)
+      (f32.sub
+        (f32.const 0x1.8p+0)
+        (f32.mul
+          (get_local $1)
+          (f32.mul
+            (get_local $0)
+            (get_local $1)))))
+   )
+```
+
+And here's the proposed text syntax:
+
+```
+   function $Q_rsqrt ($0:f32) : (f32) {
+     var $1:f32;
+     $1 = f32.reinterpret/i32(1597463007 - ((i32.reinterpret/f32($0)) >> 1));
```

Comment:
Could some of the cast operators also take advantage of the simple overloading, so this could be `$1 = f32(...)`